### PR TITLE
DSPDC-1059 Map df_ columns

### PR DESCRIPTION
**Pull request by danxmoran**
Somewhat satisfying that there's more lines of code in the schema JSON than the Scala mapping logic

### CODE REVIEW

File: schema/src/main/jade-fragments/hles_dog_diet.fragment.json [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+    {
+      "name": "df_treats_homemade_protein",
+      "datatype": "boolean"
+    },
+    {
+      "name": "df_treats_homemade_biscuits",
+      "datatype": "boolean"
+    },
+    {
+      "name": "df_treats_pumpkins",
```
Comment (snwessel) [inline schema/src/main/jade-fragments/hles_dog_diet.fragment.json:169]:
```suggestion
      "name": "df_treats_pumpkin",
```
Was this intentional? I'm just comparing things to the mapping sheet and noticed the difference.

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      dfInfrequentSupplementsAlkalinize =
+        rawRecord.getOptionalNumber("df_s_alkalinize_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsAcidify =
+        rawRecord.getOptionalNumber("df_s_acidify_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsTaurine =
+        rawRecord.getOptionalNumber("df_s_taurine_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsAntiox =
+        rawRecord.getOptionalNumber("df_s_antiox_ltd").filter(_ != 0L),
+      dfInfrequentSupplementsCoenzymeQ10 = rawRecord.getOptionalNumber("df_s_q10_ltd"),
+      dfInfrequentSupplementsOther = otherSupps,
```
Comment (snwessel) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala:219]: Should 0 be filtered out of the "other" and "Coenzyme Q10" fields too?

File: schema/src/main/jade-fragments/hles_dog_diet.fragment.json (hunk 1 above)
Comment (danxmoran) [reply schema/src/main/jade-fragments/hles_dog_diet.fragment.json:169]: Good catch

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala (hunk 2 above)
Comment (danxmoran) [reply hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/DietTransformations.scala:219]: It's filtered out of "other" at the `val` declaration, but good catch on the Q10

**Pull request merged**